kaldb/src/test/java/com/slack/kaldb/logstore/FieldConflictsTest.java

```diff
@@ -0,0 +1,227 @@
+package com.slack.kaldb.logstore;
+
+import static com.slack.kaldb.testlib.TemporaryLogStoreAndSearcherRule.findAllMessages;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import brave.Tracing;
+import com.slack.kaldb.testlib.MessageUtil;
+import com.slack.kaldb.testlib.TemporaryLogStoreAndSearcherRule;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class FieldConflictsTest {
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    Tracing.newBuilder().build();
+  }
+
+  @Rule
+  public TemporaryLogStoreAndSearcherRule strictLogStore =
+      new TemporaryLogStoreAndSearcherRule(false);
+
+  public FieldConflictsTest() throws IOException {}
```

Comment:
Unused constructor?

Reply:
Need since we throw a checked exception in @Rule.